### PR TITLE
Fix handling cues with missing end times

### PR DESCRIPTION
**Pull request by hghazzi**
When cues do not have end times, the start time of the following cue,
that does not have the same start time, should be used.

JW7-2854

### CODE REVIEW

File: src/js/view/captionsrenderer.js [hunk 1]
```diff
@@ -102,9 +102,26 @@ define([
...
+            function getEndTime(cue, i) {
+                if (cue.endTime ||  _.last(allCues) === cue) {
+                    return cue.endTime;
+                }
+
+                if (cue.startTime < allCues[i+1].startTime) {
+                    return allCues[i+1].startTime;
+                }
+
+                return getEndTime(cue, i+1);
```
Comment (hghazzi) [inline src/js/view/captionsrenderer.js]: If using recursion here is a concern, I can make it iterative. It just seemed a lot cleaner this way.

File: test/unit/captionsrenderer-test.js [hunk 2]
```diff
@@ -21,9 +21,45 @@ define([
...
+        var currentNumCues = [1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 2, 1, 2, 4, 4, 1, 1, 1, 1, 0, 0, 0];
+
+        for (var i = 0; i < currentNumCues.length; i += 1) {
+            assert.equal(captionsRenderer.getCurrentCues(allCues, i).length, currentNumCues[i], 'Invalid number of cues at position ' + i);
+        }
+    });
+
+    test('should show the correct number of cues at any given position in time with some missing end times', function (assert) {
+        var allCues = [
+            new VTTCue(0, undefined, 'HG: Morning, Rob.'),
```
Comment (egreaves) [inline test/unit/captionsrenderer-test.js]: @hghazzi using `undefined` throws a TypeError in Chrome and sets `endTime` to `NaN` in Safari. A more realistic test is just using 0 (or `null`, which also sets `endTime` to 0 in Chrome and Safari), as that was the case in the stream where the bug was first discovered. Also, our parser returns 0 for invalid input.

**Comment by robwalch** [comment]
Fix the parsing issue in Flash https://github.com/jwplayer/jwplayer-commercial/blob/master/src/flash/com/longtailvideo/jwplayer/parsers/SRT.as#L55

I don't want to add work arounds here.

**Comment by hghazzi** [comment]
@robwalch Do you think it might still be worth keeping this code as part of the fix to handle any potential edge case where end times are no available? Or should we maybe throw an exception since there should always be an end time?

**Comment by robwalch** [comment]
Throwing an exception only in the debug player would be ok. Release builds should fail silently. There are examples of this in different parts of the player - See utils.tryCatch

I'd rather not hide the problem even for edge cases.
